install.sh

```diff
@@ -916,7 +923,15 @@ ohai "Downloading and installing Homebrew..."
     fi
   fi
 
-  if [[ ! -d "${HOMEBREW_CORE}" ]]
+  if [[ -n "${HOMEBREW_INSTALL_FROM_API-}" ]]
+  then
+    ohai 'Skip tapping homebrew/core because `$HOMEBREW_INSTALL_FROM_API` is set.'
+    # Unset HOMEBREW_DEVELOPER since it is no longer needed and causes warnings during brew update below
```

Comment:
What warnings? Was it set by this script or elsewhere?

Reply:
`$HOMEBREW_DEVELOPER` may be set here:

https://github.com/Homebrew/install/blob/a64608bec0e11adfae1cea3b6b57fd4dcd3b1b69/install.sh#L482-L487

And the installation script would proceed to do a `brew update`, which will warn if both `$HOMEBREW_DEVELOPER` and `$HOMEBREW_INSTALL_FROM_API` are set:

https://github.com/Homebrew/install/blob/a64608bec0e11adfae1cea3b6b57fd4dcd3b1b69/install.sh#L938

```
...snip...
HEAD is now at b1577d347 Merge pull request #13038 from gromgit/github_api/fix_url
==> Skip tapping homebrew/core because `$HOMEBREW_INSTALL_FROM_API` is set.
Warning: Developers should not have HOMEBREW_INSTALL_FROM_API set!
Please unset HOMEBREW_INSTALL_FROM_API or turn developer mode off by running:
  brew developer off

Warning: Developers should not have HOMEBREW_INSTALL_FROM_API set!
Please unset HOMEBREW_INSTALL_FROM_API or turn developer mode off by running:
  brew developer off

Warning: Developers should not have HOMEBREW_INSTALL_FROM_API set!
Please unset HOMEBREW_INSTALL_FROM_API or turn developer mode off by running:
  brew developer off

==> Downloading https://ghcr.io/v2/homebrew/portable-ruby/portable-ruby/blobs/sha256:cd7fffb18ef9338baa670fc5e8fce99b0e3cc0f0fd7627bcbb56f3c8d54161d4
...snip...
```